packages/puppeteer-core/src/bidi/CDPSession.ts

```diff
@@ -86,10 +88,13 @@ export class BidiCdpSession extends CDPSession {
     if (this.#connection === undefined || this.#detached) {
       return;
     }
+
     try {
-      await this.frame.client.send('Target.detachFromTarget', {
-        sessionId: this.id(),
-      });
+      if (this.#userCreated) {
+        await this.frame.client.send('Target.detachFromTarget', {
+          sessionId: this.id(),
+        });
+      }
     } finally {
       BidiCdpSession.sessions.delete(this.id());
       this.#detached = true;
```

Comment:
can we learn the mapper closed the session and set the detached flag? I would rather suppress the error than introduce the notion of the user created sessions

Reply:
We can just not call it. As the this is done on `browsingContext.constextDestroyed` event, and let the user handle the CDP session as they please.

Reply:
We probably only need the BidiCdpSession.sessions.delete bit then

Reply:
OK, I will update the PR and title